File: launch/interface/src/main/java/xsbti/ScalaProvider.java
package xsbti;

import java.io.File;

/** Provides access to the jars and classes for a particular version of Scala.*/
public interface ScalaProvider
{
	public Launcher launcher();
	/** The version of Scala this instance provides.*/
	public String version();

	/** A ClassLoader that loads the classes from scala-library.jar and scala-compiler.jar.*/
	public ClassLoader loader();
	/** Returns the scala-library.jar and scala-compiler.jar for this version of Scala. */
	public File[] jars();
	public File libraryJar();
	public File compilerJar();
	/** Creates an application provider that will use 'loader()' as the parent ClassLoader for
	* the application given by 'id'.  This method will retrieve the application if it has not already
	* been retrieved.*/
	public AppProvider app(ApplicationID id);
}

File: sbt-launcher/src/main/scala-2.11/coursier/sbtlauncher/ScalaProvider.scala
package coursier.sbtlauncher

import java.io.File

final case class ScalaProvider(
  launcher: xsbti.Launcher,
  version: String,
  loader: ClassLoader,
  jars: Array[File],
  libraryJar: File,
  compilerJar: File,
  createApp: xsbti.ApplicationID => xsbti.AppProvider
) extends xsbti.ScalaProvider {
  def app(id: xsbti.ApplicationID): xsbti.AppProvider =
    createApp(id)
}
